app.py

```diff
@@ -9,43 +11,80 @@
 parser.add_argument('name')
 parser.add_argument('function')
 
+params_schema = Schema({'name':str,'function':str})
+id_schema = Schema(All(str,Length(min=24,max=24,),Match(r'^[0123456789abcdef]*$')))
 
 class Thing(Resource):
-    def get(self, name):
-        thing = Model.get_by_name(name)
+    def get(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
+        try:
+            thing = Model.get_by_id(_id)
+        except:
+            abort(500, message='Unexpected Error '+str(sys.exc_info()[0]))
         if not thing:
-            abort(404, message='Thing {} not found'.format(name))
-        return vars(thing)
+            abort(404, message='Thing {} not found'.format(_id))
+        return jsonify(vars(thing))
 
-    def delete(self, name):
-        thing = Model.delete(name)
-        if not thing:
-            abort(404, message='Thing {} not found'.format(name))
-        return vars(thing), 204
+    def delete(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
+        try:
+            response = Model.delete(_id)
+        except:
+            abort(500, message='Unexpected Error '+str(sys.exc_info()[0]))
+        return jsonify(response)
 
-    def put(self, name):
+    def put(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
         args = parser.parse_args()
-        thing = Model.update(name, args['function'])
-        if not thing:
-            abort(404, message='Thing {} not found'.format(name))
-        return vars(thing), 201
+        try:
+            response = Model.update(_id, args['name'], args['function'])
+        except:
+            abort(500, message='Unexpected Error '+str(sys.exc_info()[0]))
+        return jsonify(response)
 
 class ThingList(Resource):
     def get(self):
-        thing_list = list(Model.find_all())
-        thing_dict = {}
-        for t in thing_list:
-            thing_dict[str(t['_id'])] = {'name':t['name'],'function':t['function']}
-        return thing_dict
+        args = parser.parse_args()
+        params = {}
+        if args['name']:
+            params['name'] = args['name']
+        if args['function']:
+            params['funtion'] = args['function']
+        try:
+            params_schema(params)
+        except:
+            abort(400,message='invalid parameter')
+        try:
+            thing_list = Model.get_by_params(params)
+        except:
+            abort(500, message='Unexpected Error '+str(sys.exc_info()[0]))
+        thing_dict = dict(map(lambda t: (t._id, {'name':t.name, 'function': t.function}),
```

Comment:
what are you doing here? just serialize the entire list of objects into json dictionaries and return the list of data with a status code